onnxruntime/core/providers/nnapi/nnapi_builtin/builders/model_builder.cc

```diff
@@ -543,17 +543,19 @@ Status ModelBuilder::Compile(std::unique_ptr<Model>& model) {
 
     bool all_ops_supported = std::all_of(supported_ops, supported_ops + num_nnapi_ops_,
                                          [](bool is_supported) { return is_supported; });
-    // TODO, To allow fallback to CPU if it's not strict CPU_DISABLED mode
+    // To allow fallback to CPU if it's not strict CPU_DISABLED mode
     if (!all_ops_supported) {
+      if (target_device_option_ != TargetDeviceOption::CPU_DISABLED_SOFT) {
```

Comment:
so if target device option is CPU_DISABLED_SOFT, `device_handles` will not include the CPU device and `ANeuralNetworksCompilation_createForDevices` will be called. how will it fall back to CPU?

Reply:
Thanks for catching. I should add the `else` branch back.